### PR TITLE
Add kinnews_kirnews

### PR DESCRIPTION
**Pull request by saradhix**
Add kinnews and kirnews

### CODE REVIEW

File: datasets/kinnews_kirnews/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+Each dataset is in comma-separated-value (csv) format, with columns that are described bellow (Note that in the cleaned versions we only remain with 'label','title', and 'content' columns):
+| Field | Description |
+| ----- | ----------- |
+| label | Numerical labels that range from 1 to 14 |
+| en_label | English labels |
+| kin_label | Kinyarwanda labels |
+| kir_label | Kirundi labels |
+| url | The link to the news source |
+| title | The title of the news article |
+| content | The full content of the news article |
```
Comment (lhoestq) [inline datasets/kinnews_kirnews/README.md:88]: here we expect one example from the dataset

You can get one by using `load_dataset` and then taking the first one with `dataset["train"][0]` for example

File: datasets/kinnews_kirnews/README.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+- **Leaderboard:** NA
+- **Point of Contact:** [Rubungo Andre Niyongabo1](mailto:niyongabor.andre@std.uestc.edu.cn)
+
+### Dataset Summary
+
+Kinyarwanda and Kirundi news classification datasets (KINNEWS and KIRNEWS,respectively), which were both collected from Rwanda and Burundi news websites and newspapers, for low-resource monolingual and cross-lingual multiclass classification tasks.
+
+### Supported Tasks and Leaderboards
+
+NA
```
Comment (lhoestq) [inline datasets/kinnews_kirnews/README.md:69]: Can you describe the classification task of this dataset here ?

File: datasets/kinnews_kirnews/README.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+Lang| Train | Test |
+|---| ----- | ---- |
+|Kinyarwanda|17014|4254|
+|Kirundi|3689|923|
+
+## Dataset Creation
+
+### Curation Rationale
+
+[Needs More Information]
```
Comment (lhoestq) [inline datasets/kinnews_kirnews/README.md:114]:
```suggestion
[More Information Needed]
```

Can you use `[More Information Needed]` instead ?
That's what we're using in all other dataset cards

File: datasets/kinnews_kirnews/kinnews_kirnews.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        datasets.BuilderConfig(name="kinnews_cleaned", description="Cleaned dataset for Kinyarwanda language"),
+        datasets.BuilderConfig(name="kirnews_raw", description="Dataset for Kirundi language"),
+        datasets.BuilderConfig(name="kirnews_cleaned", description="Cleaned dataset for Kirundi language"),
+    ]
+
+    def _info(self):
+        if "raw" in self.config.name:
+            features = datasets.Features(
+                {
+                    "label": datasets.Value("int32"),
```
Comment (lhoestq) [inline datasets/kinnews_kirnews/kinnews_kirnews.py:63]: Can you use the ClassLabel feature type here please ?

For example
```
"label": datasets.ClassLabel(names=["politic", "sport", etc.])
```

Also the label ids must start at zero, however it looks like the first label is politic with label `1`. So you would need to do `label = label - 1`

File: datasets/kinnews_kirnews/kinnews_kirnews.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                    "en_label": datasets.Value("string"),
+                    "url": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "content": datasets.Value("string"),
+                }
+            )
+        else:
+            features = datasets.Features(
+                {
+                    "label": datasets.Value("int32"),
```
Comment (lhoestq) [inline datasets/kinnews_kirnews/kinnews_kirnews.py:74]: same here

**Review by lhoestq — COMMENTED** [review]
Awesome thank you !

I left a few comments.

After the change to ClassLabel you'll need to regenerate the dataset_infos.json

File: datasets/kinnews_kirnews/README.md (hunk 1 above)
Comment (saradhix) [reply datasets/kinnews_kirnews/README.md:88]: Gave an example from the dataset

File: datasets/kinnews_kirnews/README.md (hunk 2 above)
Comment (saradhix) [reply datasets/kinnews_kirnews/README.md:69]: Described the 14-class classification task.

File: datasets/kinnews_kirnews/README.md (hunk 3 above)
Comment (saradhix) [reply datasets/kinnews_kirnews/README.md:114]: Changed the string to "More Information Needed" consistently

File: datasets/kinnews_kirnews/kinnews_kirnews.py (hunk 4 above)
Comment (saradhix) [reply datasets/kinnews_kirnews/kinnews_kirnews.py:63]: Yes, used ClassLabel with descriptive names.

File: datasets/kinnews_kirnews/kinnews_kirnews.py (hunk 5 above)
Comment (saradhix) [reply datasets/kinnews_kirnews/kinnews_kirnews.py:74]: Yes, used ClassLabel with descriptive names.

File: datasets/kinnews_kirnews/README.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+### Languages
+
+Kinyarwanda and Kirundi
+
+## Dataset Structure
+
+### Data Instances
+
+| Field | Value |
```
Comment (lhoestq) [inline datasets/kinnews_kirnews/README.md:92]:
```suggestion
Here is an example from the dataset:

| Field | Value |
```

File: datasets/kinnews_kirnews/README.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+### Data Fields
+
+The raw version of the data for Kinyarwanda language consists of these fields
+-label: The category of the news article
+-kin_label/kir_label: The associated label in Kinyarwanda/Kirundi language
+-en_label: The associated label in English
+-url: The URL of the news article
+-title: The title of the news article
+-content: The content of the news article
```
Comment (lhoestq) [inline datasets/kinnews_kirnews/README.md:112]:
```suggestion
- label: The category of the news article
- kin_label/kir_label: The associated label in Kinyarwanda/Kirundi language
- en_label: The associated label in English
- url: The URL of the news article
- title: The title of the news article
- content: The content of the news article
```

File: datasets/kinnews_kirnews/README.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+-economy
+-health
+-entertainment
+-history
+-technology
+-culture
+-religion
+-environment
+-education
+-relationship
```
Comment (lhoestq) [inline datasets/kinnews_kirnews/README.md:81]:
```suggestion
This dataset can be used for text classification of news articles in Kinyarwadi and Kirundi languages. Each news article can be classified into one of the 14 possible classes. The classes are:

- politics
- sport
- economy
- health
- entertainment
- history
- technology
- culture
- religion
- environment
- education
- relationship
```

**Review by lhoestq — APPROVED** [review]
Looks all good now thank you !

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Pull request merged**